### DISCUSSION TITLE
umijs从3.5升级到4.0.32

### DISCUSSION THREAD

**Discussion by gaohongwei** [umijs/umi#9822] [Q&A]
umi 升级错误, 求助
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ant-design-pro@5.0.0-beta.0
npm ERR! Found: umi@4.0.32
npm ERR! node_modules/umi
npm ERR!   umi@"^4.0.32" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer umi@"3.x" from @umijs/preset-react@2.1.6
npm ERR! node_modules/@umijs/preset-react
npm ERR!   dev @umijs/preset-react@"^2.1.6" from the root project

package.json
···
  "dependencies": {
    "@ant-design/icons": "^4.0.0",
    "@ant-design/pro-card": "^1.11.8",
    "@ant-design/pro-descriptions": "^1.8.7",
    "@ant-design/pro-form": "^1.2.0",
    "@ant-design/pro-layout": "^6.4.19",
    "@ant-design/pro-list": "^1.7.3",
    "@ant-design/pro-table": "^2.3.3",
    "@tanstack/react-query": "^4.2.3",
    "@umijs/route-utils": "^1.0.33",
    "@webscopeio/react-textarea-autocomplete": "^4.9.0",
    "antd": "^4.9.0",
    "axios": "^0.21.0",
    "classnames": "^2.2.6",
    "debug": "^4.3.1",
    "dompurify": "^2.2.2",
    "eslint-config-airbnb": "^19.0.4",
    "lodash": "^4.17.11",
    "mathjs": "^9.4.4",
    "moment": "^2.25.3",
    "omit.js": "^2.0.2",
    "plotly.js": "^1.58.4",
    "qs": "^6.9.0",
    "query-string": "^7.0.0",
    "rc-util": "^5.24.5",
    "react": "^17.0.2",
    "react-device-detect": "^2.1.2",
    "react-dom": "^17.0.2",
    "react-helmet-async": "^1.0.4",
    "react-iframe": "^1.8.0",
    "react-json-editor-ajrm": "^2.5.13",
    "react-json-view": "^1.21.3",
    "react-plotly.js": "^2.5.1",
    "react-slick": "^0.28.1",
    "regression": "^2.0.1",
    "slick-carousel": "^1.8.1",
    "umi": "^3.5.0",
    "umi-request": "^1.0.8",
    "zustand": "^4.0.0-rc.1"
  },
  "devDependencies": {
    "@ant-design/pro-cli": "^2.0.2",
    "@types/classnames": "^2.2.7",
    "@types/express": "^4.17.0",
    "@types/history": "^4.7.2",
    "@types/jest": "^26.0.0",
    "@types/lodash": "^4.14.144",
    "@types/qs": "^6.5.3",
    "@types/react": "^16.9.17",
    "@types/react-dom": "^16.8.4",
    "@types/react-helmet": "^6.1.0",
    "@typescript-eslint/eslint-plugin": "^5.20.0",
    "@typescript-eslint/parser": "^5.20.0",
    "@umijs/fabric": "^2.3.0",
    "@umijs/plugin-blocks": "^2.0.5",
    "@umijs/plugin-esbuild": "^1.0.0-beta.2",
    "@umijs/preset-ant-design-pro": "^1.2.0",
    "@umijs/preset-react": "^1.4.8",
    "@umijs/yorkie": "^2.0.3",
    "carlo": "^0.9.46",
    "cross-env": "^7.0.0",
    "cross-port-killer": "^1.1.1",
    "detect-installer": "^1.0.1",
    "enzyme": "^3.11.0",
    "eslint": "^7.32.0",
    "eslint-plugin-react": "^7.29.4",
    "eslint-plugin-react-hooks": "^4.4.0",
    "express": "^4.17.1",
    "gh-pages": "^3.0.0",
    "jsdom-global": "^3.0.2",
    "lint-staged": "^10.0.0",
    "mockjs": "^1.0.1-beta3",
    "prettier": "^2.0.1",
    "pro-download": "1.0.1",
    "puppeteer-core": "^5.0.0",
    "stylelint": "^13.0.0",
    "typescript": "^4.0.3"
  },
···

**Comment by fz6m** [comment]
https://umijs.org/docs/introduce/upgrade-to-umi-4

以前的插件现在都集成到 max 里了，需要移除这些旧的插件依赖。

你的配置里也不需要配置这些旧的插件了，给配置看看。

或者创建一个 max 新项目，一点一点挪。
